{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "── Attaching packages ─────────────────────────────────────── tidyverse 1.2.1 ──\n",
      "✔ ggplot2 3.1.0     ✔ purrr   0.2.5\n",
      "✔ tibble  2.0.1     ✔ dplyr   0.7.8\n",
      "✔ tidyr   0.8.2     ✔ stringr 1.3.1\n",
      "✔ readr   1.1.1     ✔ forcats 0.3.0\n",
      "Warning message:\n",
      "“package ‘tibble’ was built under R version 3.5.2”── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "✖ dplyr::filter() masks stats::filter()\n",
      "✖ dplyr::lag()    masks stats::lag()\n",
      "Loading required namespace: BiocManager\n",
      "Bioconductor version 3.8 (BiocManager 1.30.3), R 3.5.1 (2018-07-02)\n",
      "Update old packages: 'annotate', 'backports', 'BiocInstaller', 'BiocManager',\n",
      "  'BiocParallel', 'Biostrings', 'biovizBase', 'broom', 'callr', 'caTools',\n",
      "  'checkmate', 'circlize', 'class', 'clipr', 'clusterProfiler', 'codetools',\n",
      "  'cowplot', 'curl', 'data.table', 'dbplyr', 'DESeq', 'DOSE', 'dplyr',\n",
      "  'EDASeq', 'edgeR', 'ELMER', 'ensembldb', 'evaluate', 'forcats', 'foreign',\n",
      "  'formatR', 'gdsfmt', 'GenomeInfoDb', 'GenomicAlignments', 'GenomicFeatures',\n",
      "  'gganimate', 'ggforce', 'ggpubr', 'ggsignif', 'ggthemes', 'git2r', 'glue',\n",
      "  'gplots', 'grImport', 'Gviz', 'haven', 'Hmisc', 'htmlTable', 'httpuv',\n",
      "  'httr', 'igraph', 'imager', 'IRdisplay', 'IRkernel', 'jomo', 'knitr',\n",
      "  'later', 'lattice', 'lazyeval', 'limma', 'lme4', 'markdown', 'MASS',\n",
      "  'Matrix', 'mclust', 'mgcv', 'mice', 'mitml', 'modelr',\n",
      "  'MultiAssayExperiment', 'mvtnorm', 'openssl', 'ordinal', 'pathview',\n",
      "  'pbapply', 'processx', 'ps', 'purrr', 'quantreg', 'qvalue', 'R.utils', 'R6',\n",
      "  'RCircos', 'Rcpp', 'RcppEigen', 'RCurl', 'readr', 'readxl', 'registry',\n",
      "  'repr', 'rmarkdown', 'Rsamtools', 'rstudioapi', 'rtracklayer', 'rvcheck',\n",
      "  'stringi', 'stringr', 'sva', 'TCGAbiolinks', 'TCGAutils', 'tibble', 'tidyr',\n",
      "  'tinytex', 'tweenr', 'units', 'urltools', 'VariantAnnotation', 'xfun', 'XML'\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    combine, intersect, setdiff, union\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colMeans,\n",
      "    colnames, colSums, dirname, do.call, duplicated, eval, evalq,\n",
      "    Filter, Find, get, grep, grepl, intersect, is.unsorted, lapply,\n",
      "    lengths, Map, mapply, match, mget, order, paste, pmax, pmax.int,\n",
      "    pmin, pmin.int, Position, rank, rbind, Reduce, rowMeans, rownames,\n",
      "    rowSums, sapply, setdiff, sort, table, tapply, union, unique,\n",
      "    unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "Loading required package: stats4\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    first, rename\n",
      "\n",
      "The following object is masked from ‘package:tidyr’:\n",
      "\n",
      "    expand\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Attaching package: ‘IRanges’\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    collapse, desc, slice\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    reduce\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "Loading required package: GenomicRanges\n",
      "Loading required package: Biostrings\n",
      "Loading required package: XVector\n",
      "\n",
      "Attaching package: ‘XVector’\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    compact\n",
      "\n",
      "\n",
      "Attaching package: ‘Biostrings’\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    strsplit\n",
      "\n",
      "Loading required package: rtracklayer\n"
     ]
    }
   ],
   "source": [
    "#install.packages(\"tidyverse\")\n",
    "library(tidyverse)\n",
    "library(modelr)\n",
    "# Data Handling\n",
    "\n",
    "if (!requireNamespace(\"BiocManager\"))\n",
    "    install.packages(\"BiocManager\")\n",
    "BiocManager::install()\n",
    "library(BiocManager)\n",
    "# General bioinformatics packages\n",
    "\n",
    "library(BSgenome)\n",
    "library(BSgenome.Hsapiens.UCSC.hg19)\n",
    "# Human genome build GChr37\n",
    "\n",
    "library(biomaRt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 263,
   "metadata": {},
   "outputs": [],
   "source": [
    "source(\"TMB_Funcs.txt\")\n",
    "# My Stuff!\n",
    "\n",
    "path_name <- \"/Volumes/CCG8/processed_data\"\n",
    "genome <- BSgenome.Hsapiens.UCSC.hg19\n",
    "ensembl = useMart(\"ensembl\", dataset = \"hsapiens_gene_ensembl\")\n",
    "ensembl37 = useMart(host = \"grch37.ensembl.org\", biomart = \"ENSEMBL_MART_ENSEMBL\", dataset = \"hsapiens_gene_ensembl\")\n",
    "select <- dplyr::select"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Loading Data (Which we think is WGS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"sclc_cancercell_gardner_2017\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_character(),\n",
      "  Entrez_Gene_Id = col_integer(),\n",
      "  Chromosome = col_integer(),\n",
      "  Start_Position = col_integer(),\n",
      "  End_Position = col_integer(),\n",
      "  t_depth = col_integer(),\n",
      "  t_ref_count = col_integer(),\n",
      "  t_alt_count = col_integer(),\n",
      "  ALLELE_NUM = col_integer(),\n",
      "  DISTANCE = col_integer(),\n",
      "  STRAND_VEP = col_integer(),\n",
      "  HGNC_ID = col_integer(),\n",
      "  PUBMED = col_number(),\n",
      "  PICK = col_integer(),\n",
      "  HGVS_OFFSET = col_integer(),\n",
      "  MINIMISED = col_integer(),\n",
      "  ExAC_AF = col_double(),\n",
      "  ExAC_AF_AFR = col_double(),\n",
      "  ExAC_AF_AMR = col_double(),\n",
      "  ExAC_AF_EAS = col_double(),\n",
      "  ExAC_AF_FIN = col_double()\n",
      "  # ... with 5 more columns\n",
      ")\n",
      "See spec(...) for full column specifications.\n",
      "Warning message in rbind(names(probs), probs_f):\n",
      "“number of columns of result is not a multiple of vector length (arg 1)”Warning message:\n",
      "“2219 parsing failures.\n",
      "row # A tibble: 5 x 5 col     row col      expected   actual    file                                       expected   <int> <chr>    <chr>      <chr>     <chr>                                      actual 1   639 <NA>     120 colum… 84 colum… '/Volumes/CCG8/processed_data/sclc_cancer… file 2  3507 <NA>     120 colum… 84 colum… '/Volumes/CCG8/processed_data/sclc_cancer… row 3  3642 Chromos… an integer HG79_PAT… '/Volumes/CCG8/processed_data/sclc_cancer… col 4  3643 Chromos… an integer HG79_PAT… '/Volumes/CCG8/processed_data/sclc_cancer… expected 5  3820 Chromos… an integer HG1211_P… '/Volumes/CCG8/processed_data/sclc_cancer…\n",
      "... ................. ... ................................................................................ ........ ................................................................................ ...... ................................................................................ .... ................................................................................ ... ................................................................................ ... ................................................................................ ........ ................................................................................\n",
      "See problems(...) for more details.\n",
      "”Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"sclc_clcgp\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_character(),\n",
      "  Start_Position = col_integer(),\n",
      "  End_Position = col_integer(),\n",
      "  Protein_position = col_integer(),\n",
      "  Hotspot = col_integer()\n",
      ")\n",
      "See spec(...) for full column specifications.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"sclc_jhu\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_character(),\n",
      "  Start_Position = col_integer(),\n",
      "  End_Position = col_integer(),\n",
      "  Tumor_Sample_Barcode = col_integer(),\n",
      "  Protein_position = col_integer(),\n",
      "  Hotspot = col_integer()\n",
      ")\n",
      "See spec(...) for full column specifications.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 42\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"sclc_ucologne_2015\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_character(),\n",
      "  Entrez_Gene_Id = col_integer(),\n",
      "  Start_Position = col_integer(),\n",
      "  End_Position = col_integer(),\n",
      "  t_ref_count = col_integer(),\n",
      "  t_alt_count = col_integer(),\n",
      "  Protein_position = col_integer(),\n",
      "  Hotspot = col_integer()\n",
      ")\n",
      "See spec(...) for full column specifications.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 110\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  X1 = col_character()\n",
      ")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 158\n"
     ]
    }
   ],
   "source": [
    "data <- cbio_getData(path_name, \"sclc\", test = FALSE) %>% \n",
    "    filter(study == \"sclc_ucologne_2015\") %>% \n",
    "    select(-study)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 266,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "                                                                      \r"
     ]
    }
   ],
   "source": [
    "n <- 1000\n",
    "gl_check <- full_join((data %>% \n",
    "    mutate(cr_value = paste0(Chromosome, \":\", Start_Position, \":\", End_Position)))[1:n,],\n",
    "         (data %>% \n",
    "    pull(Hugo_Symbol))[1:n] %>% \n",
    "    {getBM(c(\"hgnc_symbol\", \"chromosome_name\", \"start_position\", \"end_position\"), filters = c(\"hgnc_symbol\"), values = ., mart = ensembl37)} %>% \n",
    "    filter(chromosome_name %in% c(1:22, \"X\", \"Y\")) %>% \n",
    "    mutate(gene_start = start_position, gene_end = end_position, Hugo_Symbol = hgnc_symbol) %>% \n",
    "    select(-c(start_position, end_position, hgnc_symbol)),\n",
    "    by = \"Hugo_Symbol\") %>% \n",
    "    mutate(in_stated_gene = Start_Position >= gene_start & End_Position <= gene_end) %>% \n",
    "    select(Hugo_Symbol, Chromosome, Start_Position, in_stated_gene, End_Position, gene_start, gene_end, chromosome_name, Tumor_Sample_Barcode, cr_value) %>% \n",
    "    distinct() %>% \n",
    "    mutate(biomart_gene = map(cr_value, ~getBM(\"hgnc_symbol\", \"chromosomal_region\", .x, ensembl37))) %>% \n",
    "    select(Hugo_Symbol, in_stated_gene, biomart_gene)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 359,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th scope=col>Hugo_Symbol</th><th scope=col>in_stated_gene</th><th scope=col>biomart_gene</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><td>PLOD1   </td><td>TRUE    </td><td>PLOD1   </td></tr>\n",
       "\t<tr><td>GABPB2  </td><td>TRUE    </td><td>GABPB2  </td></tr>\n",
       "\t<tr><td>CGN     </td><td>TRUE    </td><td>CGN     </td></tr>\n",
       "\t<tr><td>SLC27A3 </td><td>TRUE    </td><td>SLC27A3 </td></tr>\n",
       "\t<tr><td>MUC1    </td><td>TRUE    </td><td>MUC1    </td></tr>\n",
       "\t<tr><td>SPEN    </td><td>TRUE    </td><td>SPEN    </td></tr>\n",
       "\t<tr><td>UBR4    </td><td>TRUE    </td><td>UBR4    </td></tr>\n",
       "\t<tr><td>ETNK2   </td><td>TRUE    </td><td>ETNK2   </td></tr>\n",
       "\t<tr><td>VWA5B1  </td><td>TRUE    </td><td>VWA5B1  </td></tr>\n",
       "\t<tr><td>HSD11B1 </td><td>TRUE    </td><td>HSD11B1 </td></tr>\n",
       "\t<tr><td>GPATCH2 </td><td>TRUE    </td><td>GPATCH2 </td></tr>\n",
       "\t<tr><td>TRIM58  </td><td>TRUE    </td><td>TRIM58  </td></tr>\n",
       "\t<tr><td>TRIM58        </td><td>TRUE          </td><td>TRIM58, OR2W3 </td></tr>\n",
       "\t<tr><td>OR2T8   </td><td>TRUE    </td><td>OR2T8   </td></tr>\n",
       "\t<tr><td>OR2T12  </td><td>TRUE    </td><td>OR2T12  </td></tr>\n",
       "\t<tr><td>TMEM57  </td><td>TRUE    </td><td>TMEM57  </td></tr>\n",
       "\t<tr><td>PABPC4  </td><td>TRUE    </td><td>PABPC4  </td></tr>\n",
       "\t<tr><td>KLF17   </td><td>TRUE    </td><td>KLF17   </td></tr>\n",
       "\t<tr><td>STIL    </td><td>TRUE    </td><td>STIL    </td></tr>\n",
       "\t<tr><td>RNF207  </td><td>TRUE    </td><td>RNF207  </td></tr>\n",
       "\t<tr><td>FOXD3   </td><td>TRUE    </td><td>FOXD3   </td></tr>\n",
       "\t<tr><td>FPGT                                 </td><td>TRUE                                 </td><td>FPGT       , FPGT-TNNI3K, TNNI3K     </td></tr>\n",
       "\t<tr><td>SUFU    </td><td>TRUE    </td><td>SUFU    </td></tr>\n",
       "\t<tr><td>ADRA2A  </td><td>TRUE    </td><td>ADRA2A  </td></tr>\n",
       "\t<tr><td>A1CF          </td><td>TRUE          </td><td>ASAH2B, A1CF  </td></tr>\n",
       "\t<tr><td>LYVE1               </td><td>TRUE                </td><td>MRVI1-AS1, LYVE1    </td></tr>\n",
       "\t<tr><td>FOXR1   </td><td>TRUE    </td><td>FOXR1   </td></tr>\n",
       "\t<tr><td>INSC    </td><td>TRUE    </td><td>INSC    </td></tr>\n",
       "\t<tr><td>B4GALNT4</td><td>TRUE    </td><td>B4GALNT4</td></tr>\n",
       "\t<tr><td>B4GALNT4</td><td>TRUE    </td><td>B4GALNT4</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>IGSF1  </td><td>TRUE   </td><td>IGSF1  </td></tr>\n",
       "\t<tr><td>HTATSF1</td><td>TRUE   </td><td>HTATSF1</td></tr>\n",
       "\t<tr><td>MTMR1  </td><td>TRUE   </td><td>MTMR1  </td></tr>\n",
       "\t<tr><td>ARHGAP4</td><td>TRUE   </td><td>ARHGAP4</td></tr>\n",
       "\t<tr><td>OPN1LW </td><td>TRUE   </td><td>OPN1LW </td></tr>\n",
       "\t<tr><td>FLNA   </td><td>TRUE   </td><td>FLNA   </td></tr>\n",
       "\t<tr><td>TAZ               </td><td>TRUE              </td><td>DNASE1L1, TAZ     </td></tr>\n",
       "\t<tr><td>PLXNA3 </td><td>TRUE   </td><td>PLXNA3 </td></tr>\n",
       "\t<tr><td>CNKSR2 </td><td>TRUE   </td><td>CNKSR2 </td></tr>\n",
       "\t<tr><td>MAGEB10</td><td>TRUE   </td><td>MAGEB10</td></tr>\n",
       "\t<tr><td>DMD    </td><td>TRUE   </td><td>DMD    </td></tr>\n",
       "\t<tr><td>MXRA5  </td><td>TRUE   </td><td>MXRA5  </td></tr>\n",
       "\t<tr><td>SRPX   </td><td>TRUE   </td><td>SRPX   </td></tr>\n",
       "\t<tr><td>CASK   </td><td>TRUE   </td><td>CASK   </td></tr>\n",
       "\t<tr><td>SYN1   </td><td>TRUE   </td><td>SYN1   </td></tr>\n",
       "\t<tr><td>ZC3H12B</td><td>TRUE   </td><td>ZC3H12B</td></tr>\n",
       "\t<tr><td>PHKA1  </td><td>TRUE   </td><td>PHKA1  </td></tr>\n",
       "\t<tr><td>GPR174 </td><td>TRUE   </td><td>GPR174 </td></tr>\n",
       "\t<tr><td>ZNF711 </td><td>TRUE   </td><td>ZNF711 </td></tr>\n",
       "\t<tr><td>FAM133A</td><td>TRUE   </td><td>FAM133A</td></tr>\n",
       "\t<tr><td>SHROOM2</td><td>TRUE   </td><td>SHROOM2</td></tr>\n",
       "\t<tr><td>VCAM1  </td><td>TRUE   </td><td>VCAM1  </td></tr>\n",
       "\t<tr><td>COL11A1</td><td>TRUE   </td><td>COL11A1</td></tr>\n",
       "\t<tr><td>SYCP1  </td><td>TRUE   </td><td>SYCP1  </td></tr>\n",
       "\t<tr><td>SPAG17 </td><td>TRUE   </td><td>SPAG17 </td></tr>\n",
       "\t<tr><td>PLOD1  </td><td>TRUE   </td><td>PLOD1  </td></tr>\n",
       "\t<tr><td>ADAM30 </td><td>TRUE   </td><td>ADAM30 </td></tr>\n",
       "\t<tr><td>GJA8   </td><td>TRUE   </td><td>GJA8   </td></tr>\n",
       "\t<tr><td>IVL    </td><td>TRUE   </td><td>IVL    </td></tr>\n",
       "\t<tr><td>TTC24  </td><td>TRUE   </td><td>TTC24  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|lll}\n",
       " Hugo\\_Symbol & in\\_stated\\_gene & biomart\\_gene\\\\\n",
       "\\hline\n",
       "\t PLOD1    & TRUE     & PLOD1   \\\\\n",
       "\t GABPB2   & TRUE     & GABPB2  \\\\\n",
       "\t CGN      & TRUE     & CGN     \\\\\n",
       "\t SLC27A3  & TRUE     & SLC27A3 \\\\\n",
       "\t MUC1     & TRUE     & MUC1    \\\\\n",
       "\t SPEN     & TRUE     & SPEN    \\\\\n",
       "\t UBR4     & TRUE     & UBR4    \\\\\n",
       "\t ETNK2    & TRUE     & ETNK2   \\\\\n",
       "\t VWA5B1   & TRUE     & VWA5B1  \\\\\n",
       "\t HSD11B1  & TRUE     & HSD11B1 \\\\\n",
       "\t GPATCH2  & TRUE     & GPATCH2 \\\\\n",
       "\t TRIM58   & TRUE     & TRIM58  \\\\\n",
       "\t TRIM58         & TRUE           & TRIM58, OR2W3 \\\\\n",
       "\t OR2T8    & TRUE     & OR2T8   \\\\\n",
       "\t OR2T12   & TRUE     & OR2T12  \\\\\n",
       "\t TMEM57   & TRUE     & TMEM57  \\\\\n",
       "\t PABPC4   & TRUE     & PABPC4  \\\\\n",
       "\t KLF17    & TRUE     & KLF17   \\\\\n",
       "\t STIL     & TRUE     & STIL    \\\\\n",
       "\t RNF207   & TRUE     & RNF207  \\\\\n",
       "\t FOXD3    & TRUE     & FOXD3   \\\\\n",
       "\t FPGT                                  & TRUE                                  & FPGT       , FPGT-TNNI3K, TNNI3K     \\\\\n",
       "\t SUFU     & TRUE     & SUFU    \\\\\n",
       "\t ADRA2A   & TRUE     & ADRA2A  \\\\\n",
       "\t A1CF           & TRUE           & ASAH2B, A1CF  \\\\\n",
       "\t LYVE1                & TRUE                 & MRVI1-AS1, LYVE1    \\\\\n",
       "\t FOXR1    & TRUE     & FOXR1   \\\\\n",
       "\t INSC     & TRUE     & INSC    \\\\\n",
       "\t B4GALNT4 & TRUE     & B4GALNT4\\\\\n",
       "\t B4GALNT4 & TRUE     & B4GALNT4\\\\\n",
       "\t ⋮ & ⋮ & ⋮\\\\\n",
       "\t IGSF1   & TRUE    & IGSF1  \\\\\n",
       "\t HTATSF1 & TRUE    & HTATSF1\\\\\n",
       "\t MTMR1   & TRUE    & MTMR1  \\\\\n",
       "\t ARHGAP4 & TRUE    & ARHGAP4\\\\\n",
       "\t OPN1LW  & TRUE    & OPN1LW \\\\\n",
       "\t FLNA    & TRUE    & FLNA   \\\\\n",
       "\t TAZ                & TRUE               & DNASE1L1, TAZ     \\\\\n",
       "\t PLXNA3  & TRUE    & PLXNA3 \\\\\n",
       "\t CNKSR2  & TRUE    & CNKSR2 \\\\\n",
       "\t MAGEB10 & TRUE    & MAGEB10\\\\\n",
       "\t DMD     & TRUE    & DMD    \\\\\n",
       "\t MXRA5   & TRUE    & MXRA5  \\\\\n",
       "\t SRPX    & TRUE    & SRPX   \\\\\n",
       "\t CASK    & TRUE    & CASK   \\\\\n",
       "\t SYN1    & TRUE    & SYN1   \\\\\n",
       "\t ZC3H12B & TRUE    & ZC3H12B\\\\\n",
       "\t PHKA1   & TRUE    & PHKA1  \\\\\n",
       "\t GPR174  & TRUE    & GPR174 \\\\\n",
       "\t ZNF711  & TRUE    & ZNF711 \\\\\n",
       "\t FAM133A & TRUE    & FAM133A\\\\\n",
       "\t SHROOM2 & TRUE    & SHROOM2\\\\\n",
       "\t VCAM1   & TRUE    & VCAM1  \\\\\n",
       "\t COL11A1 & TRUE    & COL11A1\\\\\n",
       "\t SYCP1   & TRUE    & SYCP1  \\\\\n",
       "\t SPAG17  & TRUE    & SPAG17 \\\\\n",
       "\t PLOD1   & TRUE    & PLOD1  \\\\\n",
       "\t ADAM30  & TRUE    & ADAM30 \\\\\n",
       "\t GJA8    & TRUE    & GJA8   \\\\\n",
       "\t IVL     & TRUE    & IVL    \\\\\n",
       "\t TTC24   & TRUE    & TTC24  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "Hugo_Symbol | in_stated_gene | biomart_gene | \n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| PLOD1    | TRUE     | PLOD1    | \n",
       "| GABPB2   | TRUE     | GABPB2   | \n",
       "| CGN      | TRUE     | CGN      | \n",
       "| SLC27A3  | TRUE     | SLC27A3  | \n",
       "| MUC1     | TRUE     | MUC1     | \n",
       "| SPEN     | TRUE     | SPEN     | \n",
       "| UBR4     | TRUE     | UBR4     | \n",
       "| ETNK2    | TRUE     | ETNK2    | \n",
       "| VWA5B1   | TRUE     | VWA5B1   | \n",
       "| HSD11B1  | TRUE     | HSD11B1  | \n",
       "| GPATCH2  | TRUE     | GPATCH2  | \n",
       "| TRIM58   | TRUE     | TRIM58   | \n",
       "| TRIM58         | TRUE           | TRIM58, OR2W3  | \n",
       "| OR2T8    | TRUE     | OR2T8    | \n",
       "| OR2T12   | TRUE     | OR2T12   | \n",
       "| TMEM57   | TRUE     | TMEM57   | \n",
       "| PABPC4   | TRUE     | PABPC4   | \n",
       "| KLF17    | TRUE     | KLF17    | \n",
       "| STIL     | TRUE     | STIL     | \n",
       "| RNF207   | TRUE     | RNF207   | \n",
       "| FOXD3    | TRUE     | FOXD3    | \n",
       "| FPGT                                  | TRUE                                  | FPGT       , FPGT-TNNI3K, TNNI3K      | \n",
       "| SUFU     | TRUE     | SUFU     | \n",
       "| ADRA2A   | TRUE     | ADRA2A   | \n",
       "| A1CF           | TRUE           | ASAH2B, A1CF   | \n",
       "| LYVE1                | TRUE                 | MRVI1-AS1, LYVE1     | \n",
       "| FOXR1    | TRUE     | FOXR1    | \n",
       "| INSC     | TRUE     | INSC     | \n",
       "| B4GALNT4 | TRUE     | B4GALNT4 | \n",
       "| B4GALNT4 | TRUE     | B4GALNT4 | \n",
       "| ⋮ | ⋮ | ⋮ | \n",
       "| IGSF1   | TRUE    | IGSF1   | \n",
       "| HTATSF1 | TRUE    | HTATSF1 | \n",
       "| MTMR1   | TRUE    | MTMR1   | \n",
       "| ARHGAP4 | TRUE    | ARHGAP4 | \n",
       "| OPN1LW  | TRUE    | OPN1LW  | \n",
       "| FLNA    | TRUE    | FLNA    | \n",
       "| TAZ                | TRUE               | DNASE1L1, TAZ      | \n",
       "| PLXNA3  | TRUE    | PLXNA3  | \n",
       "| CNKSR2  | TRUE    | CNKSR2  | \n",
       "| MAGEB10 | TRUE    | MAGEB10 | \n",
       "| DMD     | TRUE    | DMD     | \n",
       "| MXRA5   | TRUE    | MXRA5   | \n",
       "| SRPX    | TRUE    | SRPX    | \n",
       "| CASK    | TRUE    | CASK    | \n",
       "| SYN1    | TRUE    | SYN1    | \n",
       "| ZC3H12B | TRUE    | ZC3H12B | \n",
       "| PHKA1   | TRUE    | PHKA1   | \n",
       "| GPR174  | TRUE    | GPR174  | \n",
       "| ZNF711  | TRUE    | ZNF711  | \n",
       "| FAM133A | TRUE    | FAM133A | \n",
       "| SHROOM2 | TRUE    | SHROOM2 | \n",
       "| VCAM1   | TRUE    | VCAM1   | \n",
       "| COL11A1 | TRUE    | COL11A1 | \n",
       "| SYCP1   | TRUE    | SYCP1   | \n",
       "| SPAG17  | TRUE    | SPAG17  | \n",
       "| PLOD1   | TRUE    | PLOD1   | \n",
       "| ADAM30  | TRUE    | ADAM30  | \n",
       "| GJA8    | TRUE    | GJA8    | \n",
       "| IVL     | TRUE    | IVL     | \n",
       "| TTC24   | TRUE    | TTC24   | \n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "     Hugo_Symbol in_stated_gene biomart_gene                         \n",
       "1    PLOD1       TRUE           PLOD1                                \n",
       "2    GABPB2      TRUE           GABPB2                               \n",
       "3    CGN         TRUE           CGN                                  \n",
       "4    SLC27A3     TRUE           SLC27A3                              \n",
       "5    MUC1        TRUE           MUC1                                 \n",
       "6    SPEN        TRUE           SPEN                                 \n",
       "7    UBR4        TRUE           UBR4                                 \n",
       "8    ETNK2       TRUE           ETNK2                                \n",
       "9    VWA5B1      TRUE           VWA5B1                               \n",
       "10   HSD11B1     TRUE           HSD11B1                              \n",
       "11   GPATCH2     TRUE           GPATCH2                              \n",
       "12   TRIM58      TRUE           TRIM58                               \n",
       "13   TRIM58      TRUE           TRIM58, OR2W3                        \n",
       "14   OR2T8       TRUE           OR2T8                                \n",
       "15   OR2T12      TRUE           OR2T12                               \n",
       "16   TMEM57      TRUE           TMEM57                               \n",
       "17   PABPC4      TRUE           PABPC4                               \n",
       "18   KLF17       TRUE           KLF17                                \n",
       "19   STIL        TRUE           STIL                                 \n",
       "20   RNF207      TRUE           RNF207                               \n",
       "21   FOXD3       TRUE           FOXD3                                \n",
       "22   FPGT        TRUE           FPGT       , FPGT-TNNI3K, TNNI3K     \n",
       "23   SUFU        TRUE           SUFU                                 \n",
       "24   ADRA2A      TRUE           ADRA2A                               \n",
       "25   A1CF        TRUE           ASAH2B, A1CF                         \n",
       "26   LYVE1       TRUE           MRVI1-AS1, LYVE1                     \n",
       "27   FOXR1       TRUE           FOXR1                                \n",
       "28   INSC        TRUE           INSC                                 \n",
       "29   B4GALNT4    TRUE           B4GALNT4                             \n",
       "30   B4GALNT4    TRUE           B4GALNT4                             \n",
       "⋮    ⋮           ⋮              ⋮                                    \n",
       "971  IGSF1       TRUE           IGSF1                                \n",
       "972  HTATSF1     TRUE           HTATSF1                              \n",
       "973  MTMR1       TRUE           MTMR1                                \n",
       "974  ARHGAP4     TRUE           ARHGAP4                              \n",
       "975  OPN1LW      TRUE           OPN1LW                               \n",
       "976  FLNA        TRUE           FLNA                                 \n",
       "977  TAZ         TRUE           DNASE1L1, TAZ                        \n",
       "978  PLXNA3      TRUE           PLXNA3                               \n",
       "979  CNKSR2      TRUE           CNKSR2                               \n",
       "980  MAGEB10     TRUE           MAGEB10                              \n",
       "981  DMD         TRUE           DMD                                  \n",
       "982  MXRA5       TRUE           MXRA5                                \n",
       "983  SRPX        TRUE           SRPX                                 \n",
       "984  CASK        TRUE           CASK                                 \n",
       "985  SYN1        TRUE           SYN1                                 \n",
       "986  ZC3H12B     TRUE           ZC3H12B                              \n",
       "987  PHKA1       TRUE           PHKA1                                \n",
       "988  GPR174      TRUE           GPR174                               \n",
       "989  ZNF711      TRUE           ZNF711                               \n",
       "990  FAM133A     TRUE           FAM133A                              \n",
       "991  SHROOM2     TRUE           SHROOM2                              \n",
       "992  VCAM1       TRUE           VCAM1                                \n",
       "993  COL11A1     TRUE           COL11A1                              \n",
       "994  SYCP1       TRUE           SYCP1                                \n",
       "995  SPAG17      TRUE           SPAG17                               \n",
       "996  PLOD1       TRUE           PLOD1                                \n",
       "997  ADAM30      TRUE           ADAM30                               \n",
       "998  GJA8        TRUE           GJA8                                 \n",
       "999  IVL         TRUE           IVL                                  \n",
       "1000 TTC24       TRUE           TTC24                                "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "gl_check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 288,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "1000"
      ],
      "text/latex": [
       "1000"
      ],
      "text/markdown": [
       "1000"
      ],
      "text/plain": [
       "[1] 1000"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "gl_check %>% \n",
    "    nrow()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 338,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "964"
      ],
      "text/latex": [
       "964"
      ],
      "text/markdown": [
       "964"
      ],
      "text/plain": [
       "[1] 964"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "gl_check %>% \n",
    "    mutate(biomart_gene = map(biomart_gene, ~tibble(hgnc_symbol = as.character(.x$hgnc_symbol)))) %>% \n",
    "    unnest() %>% \n",
    "    filter(Hugo_Symbol == hgnc_symbol) %>% \n",
    "    nrow()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 376,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th scope=col>hgnc_symbol</th></tr></thead>\n",
       "<tbody>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|l}\n",
       " hgnc\\_symbol\\\\\n",
       "\\hline\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "hgnc_symbol | \n",
       "||\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "     hgnc_symbol"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "getBM(\"hgnc_symbol\", c(\"chromosomal_region\"), \"1:60700000:60700100\", ensembl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.5.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
